fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DataServlet.java

```diff
@@ -172,7 +176,8 @@ public DataServlet(
             BroadcastChannel<String> preResponseStoredNotifications,
             HttpResponseMaker httpResponseMaker,
             ResponseFormatResolver formatResolver,
-            DataApiRequestFactory dataApiRequestFactory
+            DataApiRequestFactory dataApiRequestFactory,
+            ResponseProcessorFactory responseProcessorFactory
```

Comment:
We haven't been doing that for the DataServlet. And we could, but seeing as how
we always add something to this constructor whenever we need to add something
new to inject, `DataServlet` would *very* quickly become *very* noisy.

Also, although it is technically possible to override the `DataServlet`, I think
that's sufficiently weird and rare enough that having hordes of giant 
deprecated constructors running around would do more harm than good.

Especially since this is an easy enough thing to fix.